modules/shortcodes/wordads.php

```diff
@@ -53,14 +57,41 @@ static function wordads_shortcode_html( $atts, $content = '' ) {
 			return '<div>' . __( 'The WordAds module is not active', 'jetpack' ) . '</div>';
 		}
 
-		$html = '<div class="jetpack-wordad" itemscope itemtype="https://schema.org/WPAdBlock">';
-
-		$html .= '</div>';
-
+		$html = '<div class="jetpack-wordad" itemscope itemtype="https://schema.org/WPAdBlock"></div>';
 		$html = $wordads->insert_inline_ad( $html );
 
 		return $html;
 	}
+
+	public static function gutenblock_render( $attr ) {
+		global $wordads;
+
+		/** This filter is already documented in modules/wordads/wordads.php `insert_ad()` */
+		if ( is_feed() || apply_filters( 'wordads_inpost_disable', false ) ) {
+			return '';
+		}
+
+		if ( $wordads->option( 'wordads_house' ) ) {
+			return $wordads->get_ad( 'inline', 'house' );
+		}
+
+		$section_id = 0 === $wordads->params->blog_id ? WORDADS_API_TEST_ID : $wordads->params->blog_id . '6'; // 6 is to keep track of gutenblock ads
+		$align = 'center';
+		if ( isset( $attr['align'] ) && in_array( $attr['align'], array( 'left', 'center', 'right' ) ) ) {
+			$align = $attr['align'];
+		}
+		$align = 'align' . $align;
+
+		$format = 'mrec';
+		if ( isset( $attr['format'] ) && in_array( $attr['format'], array_keys( WordAds::$ad_tag_ids ) ) ) {
+			$format = $attr['format'];
+		}
+
+		$height = WordAds::$ad_tag_ids[$format]['height'];
+		$width = WordAds::$ad_tag_ids[$format]['width'];
+		$snippet = $wordads->get_ad_snippet( $section_id, $height, $width, 'inline', 'float:left;margin-right:5px;margin-top:0px;' );
+		return $wordads->get_ad_div( 'inline', $snippet, array( $align ) );
```

Comment:
In addition to alignment, the Gutenberg editor adds a class and allows the post author to add additional classes.

If we won't use those in the output (which I suspect is intended), we should disable in the editor as well.

If you'll confirm that's intended, I'll open an issue to handle that editor-side 👍

Reply:
Yeah, due to requirements from our ad partners we generally want to minimize the amount of alteration our users can perform on the ad units.

Reply:
https://github.com/Automattic/wp-calypso/pull/30668